### PR TITLE
CLDC-1852 add weekly rent period and reorder options

### PR DESCRIPTION
**Pull request by arfacamble**
https://digital.dclg.gov.uk/jira/browse/CLDC-1852

1. add in a new option for the period question and reorder existing answer options
2. ensure that the lettings log calculates values correctly that relate to the period

|image
|-

|image
|-

### CODE REVIEW

**Review by SamSeed-Softwire — COMMENTED** [review]
The stuff you've written all looks fine.

Are there other places in the codebase however which need to reference the new answer option? E.g. the following look like they might need it:
- `app/models/form/lettings/pages/care_home_weekly.rb`
- `app/models/form/lettings/pages/rent_weekly.rb`
- `app/models/form/lettings/questions/tshortfall.rb`
- `config/forms/2022_2023.json`, assuming it's relevant to 22/23

Is it worth adding some tests for the new answer option in `spec/models/lettings_log_spec.rb` as well?

**Comment by arfacamble** [comment]
> The stuff you've written all looks fine.
> 
> Are there other places in the codebase however which need to reference the new answer option? E.g. the following look like they might need it:
> 
> * `app/models/form/lettings/pages/care_home_weekly.rb`
> * `app/models/form/lettings/pages/rent_weekly.rb`
> * `app/models/form/lettings/questions/tshortfall.rb`
> * `config/forms/2022_2023.json`, assuming it's relevant to 22/23
> 
> Is it worth adding some tests for the new answer option in `spec/models/lettings_log_spec.rb` as well?

great catch

File: app/models/lettings_log.rb [hunk 1]
```diff
@@ -379,6 +379,10 @@ def receives_uc_with_housing_element_excl_housing_benefit?
     hb == 6
   end
 
+  def receives_any_housing_benefit?
```
Comment (SamSeed-Softwire) [inline app/models/lettings_log.rb]: Could this new method also be used to refactor `receives_housing_related_benefits?`? Or did you even need to create this method and could you have just used `receives_housing_related_benefits?` instead?

File: spec/fixtures/forms/2021_2022.json
Comment (SamSeed-Softwire) [inline spec/fixtures/forms/2021_2022.json:1]: Is the new answer option necessary here, given this file is for 2021/22?

File: app/models/form/lettings/pages/rent_weekly.rb [hunk 2]
```diff
@@ -4,30 +4,8 @@ def initialize(id, hsh, subsection)
...
-      { "period" => 5, "household_charge" => nil, "is_carehome" => nil },
-      { "period" => 6, "household_charge" => 0, "is_carehome" => nil },
-      { "period" => 6, "household_charge" => nil, "is_carehome" => nil },
-      { "period" => 7, "household_charge" => 0, "is_carehome" => nil },
-      { "period" => 7, "household_charge" => nil, "is_carehome" => nil },
-      { "period" => 8, "household_charge" => 0, "is_carehome" => nil },
-      { "period" => 8, "household_charge" => nil, "is_carehome" => nil },
-      { "period" => 9, "household_charge" => 0, "is_carehome" => nil },
-      { "period" => 9, "household_charge" => nil, "is_carehome" => nil },
+      { "rent_and_charges_paid_weekly?" => true, "household_charge" => 0, "is_carehome?" => false },
```
Comment (SamSeed-Softwire) [inline app/models/form/lettings/pages/rent_weekly.rb:7]: There are various other places that `"is_carehome" => nil or 0` could be replaced with `"is_carehome?" => false`, for example in `app/models/form/lettings/pages/rent_4_weekly.rb`. I'd recommend either doing that as part of this ticket, or raising a new tech debt ticket.

File: spec/fixtures/forms/2021_2022.json
Comment (arfacamble) [reply spec/fixtures/forms/2021_2022.json:1]: this is a test fixture. If I don't put it here I have to rewrite the whole lettings_log_spec and that would really stretch the scope of this ticket!

File: app/models/lettings_log.rb (hunk 1 above)
Comment (arfacamble) [reply app/models/lettings_log.rb]: just use that! good catch

File: app/models/form/lettings/pages/rent_weekly.rb (hunk 2 above)
Comment (arfacamble) [reply app/models/form/lettings/pages/rent_weekly.rb:7]: Not sure, I don't want to change half the conditions in rent_4_weekly, so then I change them all with some new methods, and then I have to search to for everywhere that household_charge is nil or 0 and everywhere period is 3 and then write new methods for the other conditions in those depends on, and so on and so on.
Where do I cut the scope of the ticket off?

Comment (arfacamble) [reply app/models/form/lettings/pages/rent_weekly.rb:7]: ok I relented on this in a big way

File: spec/fixtures/forms/2021_2022.json
Comment (SamSeed-Softwire) [reply spec/fixtures/forms/2021_2022.json:1]: Ah gotcha

File: app/models/form/lettings/pages/rent_weekly.rb (hunk 2 above)
Comment (SamSeed-Softwire) [reply app/models/form/lettings/pages/rent_weekly.rb:7]: > ok I relented on this in a big way

What do you mean?

Comment (arfacamble) [reply app/models/form/lettings/pages/rent_weekly.rb:7]: I made the changes

**Review by SamSeed-Softwire — APPROVED** [review]
The time has come!

**Pull request merged**